### PR TITLE
Let if statement match the assert statement above it.

### PR DESCRIPTION
**Pull request by ghost**
This change is worth it because should we not need the assert anymore, it will be more obvious when the time comes. The behavior is exactly the same as before.

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Text/Encoding.cs [hunk 1]
```diff
@@ -1301,9 +1301,9 @@ internal unsafe void AdjustBytes(int count)
             internal unsafe byte GetNextByte()
             {
                 Debug.Assert(_bytes < _byteEnd, "[EncodingCharBuffer.GetNextByte]Expected more date");
-                if (_bytes >= _byteEnd)
-                    return 0;
-                return *(_bytes++);
+                if (_bytes < _byteEnd)
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Text/Encoding.cs]: I do not think that this is more readable with this change. The original code used early exit for exceptional condition that is generally easier to understand.  I really like how LLVM coding guidelines describe it: https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code . This one is too small of a method to clearly see the benefits of the early exit pattern, but it is nice to be consistent.

Comment (pi1024e) [reply src/libraries/System.Private.CoreLib/src/System/Text/Encoding.cs]: However, when you refactor it, and see the if statement and the asserts are the same, this raises the question: does the assert need to be there?

Comment (jkotas) [reply src/libraries/System.Private.CoreLib/src/System/Text/Encoding.cs]: This code operates on unsafe pointers. It is not unusual to both assert and check in code like this  prone to buffer overruns for defense in depth.

Comment (pi1024e) [reply src/libraries/System.Private.CoreLib/src/System/Text/Encoding.cs]: Got it.

**Pull request closed without merging**